usaspending_api/search/tests/unit/filters/naics.py

```diff
@@ -0,0 +1,37 @@
+from usaspending_api.search.elasticsearch.filters.naics import NaicsCodes
```

Comment:
I'd also actually like to see an integration test that hits ES to be sure that: not only does it generate the expected `query_string` syntax, but that your assumptions about how ES will execute that query string on an indexed data set are correct.

Please add that, since we all make assumptions about ES that we are discovering new things about every day.

Reply:
Integration test was modified at https://github.com/fedspendingtransparency/usaspending-api/pull/2336/files#diff-53a443dd0745aa53a400aa5732f26ebd

Tons more tests could be made of course, but I was trying to respect the simple structure that existed, in order to avoid having excessive tests checking the same thing.

Reply:
Good to see we have at least one real hit of ES data. 
However, can you piggy back on that test and run a few more data scenarios / queries that force execution of some of the more convoluted query strings and assert the results are as expected. 
Since the testbed is already there, should just by a bit of copy-paste-edit-rename.

Reply:
I'd want to be a little cleaner than a copy/paste strategy, particularly with fixtures, since what we have for spending_by_award is a bit of a mess and it will only get harder to work with (and ensure that failed tests are failing for the reason we think they are) as the fixture gets built out to support more complex cases.
While cleaning up the fixture data and restructuring this test class is certainly something I'd like to do, there's a question of scope in doing so, since the way I'd want to add more integration tests is more effort than a copy/paste job.

Reply:
@BrianZito Take a look at this branch as a starter `mod/dev-4640-add-exclusion-logic-integ-tests`, that can be merged into yours. 

Should just be a few things that need to change if you build off of the fixture data that already exists.

https://github.com/fedspendingtransparency/usaspending-api/compare/mod/dev-4640-add-exclusion-logic...mod/dev-4640-add-exclusion-logic-integ-tests

Reply:
Your example shows cake pattern, which I was asked not to do before. Are we okay with adding fixture data outside python fixtures now?

Reply:
The fixture is already shared among tests. And the data created in each test will be torn down after that test since it uses a _`function`_ scope for the fixture, yes? That includes the data from the fixture, plus any modifications or additions or deletions to that data needed for your specific test.

If we wanted to isolate test data to a per-test basis, then we should not create that data as a shareable fixture, but just create it within the individual test, so there is not a ripple effect when shared test data changes. 

I'm not sure where the rule stands. If you know, then choose one of two approaches:
1) Use the fixture as a shared fixture, add what you need onto it in your test
2) Copy the fixture code into your test, add what you need onto it in your test

Either way you have a path forward.

Reply:
I was thinking to limit setup time on each test, you could have multiple asserts in one test, modifying the data for each part of the test. That way you could pack a bunch of different `require`/`exclude` combos into 1 test. But these 3 are fine.

Reply:
Keep in mind these are unit tests, whose setup takes milliseconds.

Reply:
Keep in mind that this would still require reindexing the elasticsearch each time no matter what, which I presume is where most of the time comes from.